### PR TITLE
tests: Hide GetPodsCertIfSynced

### PR DESCRIPTION
**Pull request by oshoval**
### What this PR does
Function is used only on one file, therefore move it from utils.
Remove unrequired boolean which isn't used anyhow,
the error and cert return values are enough.

Split `getPodsCertIfSynced` to:
1. Get all certs.
2. Validate certs are valid and equal.
3. Use first cert to verify it was updated.

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by oshoval** [comment]
/test pull-kubevirt-e2e-k8s-1.29-sig-compute-migrations

open https://github.com/kubevirt/kubevirt/pull/11764 to clean the error log a bit
cluster-up failed there

File: tests/infrastructure/certificates.go [hunk 1]
```diff
@@ -197,19 +202,14 @@ func getCertFromSecret(secretName string) []byte {
 	return nil
 }
 
-// getPodsCertIfSynced returns the certificate for all matching pods once all of them use the same certificate
-func getPodsCertIfSynced(labelSelector string, namespace string, port string) (cert []byte, err error) {
-	certs, err := tests.GetCertsForPods(labelSelector, namespace, port)
-	if err != nil {
-		return nil, err
-	}
+func hasSyncedCerts(certs [][]byte) []byte {
```
Comment (EdDev) [inline tests/infrastructure/certificates.go:205]: Maybe it should be called `identicalCert()`.

Prefixing `Has` hints the function returns true or false.

**Review by EdDev — APPROVED** [review]
I had one comment about the naming in the 2nd commit, but it is not a blocker.
If you'll have the time, I would suggest to change it in order to reflect better what it does.

File: tests/infrastructure/certificates.go (hunk 1 above)
Comment (xpivarc) [inline tests/infrastructure/certificates.go:205]: `hasSyncedCerts` and it returns a cert? It does not sound intuitive to me.

Comment (oshoval) [reply tests/infrastructure/certificates.go:205]: will change according https://github.com/kubevirt/kubevirt/pull/11763#discussion_r1573847961

**Comment by oshoval** [comment]
Addressed comments
Thanks

**Review by EdDev — APPROVED** [review]

**Comment by oshoval** [comment]
Updated function to return bool
verify it returns true in the caller, and if so, use the first cert to validate it was updated.

**Review by EdDev — APPROVED** [review]

**Comment by oshoval** [comment]
Hi @brianmcarey 

is this known ?
11:56:05: modprobe: FATAL: Module vfio_mdev not found in directory /lib/modules/5.14.0-284.59.1.el9_2.x86_64

https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/11763/pull-kubevirt-e2e-kind-1.27-vgpu/1785274960294449152
Thanks

**Comment by brianmcarey** [comment]
> Hi @brianmcarey
> 
> is this known ? 11:56:05: modprobe: FATAL: Module vfio_mdev not found in directory /lib/modules/5.14.0-284.59.1.el9_2.x86_64
> 
> https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/11763/pull-kubevirt-e2e-kind-1.27-vgpu/1785274960294449152 Thanks

Yes the bump kubevirtci PR should resolve this - https://github.com/kubevirt/kubevirt/pull/11807

**Comment by oshoval** [comment]
/hold

need to fix 
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/11763/pull-kubevirt-e2e-k8s-1.28-sig-compute/1785464450057244672

**Comment by oshoval** [comment]
/hold cancel

fixed

**Review by EdDev — APPROVED** [review]

**Pull request merged**